### PR TITLE
Call of Duty 2 server

### PR DESCRIPTION
**Pull request by BacaCro**
# Description

## Checklist for all submissions

* [ X ] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [ X ] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [ X ] Have you tested and reviewed your changes with confidence that everything works?
* [ X ] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

* [ X ] You verify that the start command applied does not use a shell script
  * [ X ] If some script is needed then it is part of a current yolk or a PR to add one
* [ X ] The egg was exported from the panel

## New egg Submissions

1. [ X ] Does your submission pass tests (server is connectable)?
2. [ - ] Does your egg use a custom docker image?
    * [ - ] Have you tried to use a generic image?
    * [ - ] Did you PR the necessary changes to make it work?
3. [ X ] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [ X ] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [ X ] You verify that the start command applied does not use a shell script
    * [ X ] If some script is needed then it is part of a current yolk or a PR to add one
6. [ X ] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/cod2/egg-call-of-duty2.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    "exported_at": "2024-04-05T12:31:07+02:00",
+    "name": "Call of Duty 2",
+    "author": "sebastijantkalcec@gmail.com",
+    "description": null,
+    "features": null,
+    "docker_images": {
+        "bacacro\/cod2pterodactyl:latest": "bacacro\/cod2pterodactyl:latest"
+    },
+    "file_denylist": [],
+    "startup": "export LD_LIBRARY_PATH=\/home\/container; export LD_PRELOAD=\"libcod.so\"; .\/cod2_lnxded +set sv_cracked {{IS_CRACKED}} +set sv_pure {{IS_PURE}} +set sv_punkbuster {{PUNKBUSTER}} +set dedicated {{IS_DEDICATED}} +set net_ip 0.0.0.0 +set net_port {{SERVER_PORT}} +set sv_maxclients {{MAX_PLAYERS}} +set g_gametype {{GAMETYPE}} +map {{DEFAULT_MAP}} +exec server.cfg;",
```
Comment (gOOvER) [inline game_eggs/cod2/egg-call-of-duty2.json:16]: +set sv_cracked {{IS_CRACKED}} ??? Thats not allowed here

File: game_eggs/cod2/egg-call-of-duty2.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            "container": "ghcr.io\/parkervcp\/installers:debian",
+            "entrypoint": "bash"
+        }
+    },
+    "variables": [
+        {
+            "name": "sv_cracked",
```
Comment (gOOvER) [inline game_eggs/cod2/egg-call-of-duty2.json:32]: No illegal thing

File: game_eggs/cod2/egg-call-of-duty2.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    },
+    "scripts": {
+        "installation": {
+            "script": "#!\/bin\/bash\r\n\r\napt install -y xz-utils tar\r\n\r\nmkdir -p \/mnt\/server\r\n\r\nif ! test -f cod2_lnxded; then\r\n        echo \"Downloading CoD2 Server Files\"\r\n        wget https:\/\/dl.baca-cro.com\/cod2\/cod2-lnxded-1.3-full.tar.xz\r\n        tar -xf cod2-lnxded-1.3-full.tar.xz -C \/mnt\/server\/\r\n        chmod +x \/mnt\/server\/cod2_lnxded\r\n        wget https:\/\/dl.baca-cro.com\/cod2\/server.cfg -O \/mnt\/server\/main\/server.cfg\r\n        echo \"Succesfully downloaded and extracted server files!\"\r\n  else\r\n        echo \"CoD2 Server Files already downloaded!\"\r\nfi\r\n\r\nif ! test -f libcod2_1_3_nomysql.so; then\r\n        echo \"Downloading libcod\"\r\n        wget https:\/\/dl.baca-cro.com\/cod2\/libcod2_1_3.so -O \/mnt\/server\/libcod.so\r\n        wget https:\/\/dl.baca-cro.com\/cod2\/libmysqlclient.so -O \/mnt\/server\/libmysqlclient.so.18\r\n        chmod +x \/mnt\/server\/libcod.so \/mnt\/server\/libmysqlclient.so.18\r\n        echo \"libcod successfully downloaded!\"\r\n  else\r\n        echo \"libcod already downloaded!\"\r\nfi\r\n\r\necho \"Successful installation!\"",
```
Comment (gOOvER) [inline game_eggs/cod2/egg-call-of-duty2.json:25]: https://dl.baca-cro.com 

Who is this?

**Review by gOOvER — CHANGES_REQUESTED** [review]
CoD 2 have no official Serverfiles

File: game_eggs/cod2/egg-call-of-duty2.json [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        "version": "PTDL_v2",
+        "update_url": null
+    },
+    "exported_at": "2024-04-05T12:31:07+02:00",
+    "name": "Call of Duty 2",
+    "author": "sebastijantkalcec@gmail.com",
+    "description": null,
+    "features": null,
+    "docker_images": {
+        "bacacro\/cod2pterodactyl:latest": "bacacro\/cod2pterodactyl:latest"
```
Comment (gOOvER) [inline game_eggs/cod2/egg-call-of-duty2.json:13]: not an official image.

**Review by gOOvER — COMMENTED** [review]
CoD 2 have no official Serverfiles

File: game_eggs/cod2/egg-call-of-duty2.json (hunk 3 above)
Comment (BacaCro) [reply game_eggs/cod2/egg-call-of-duty2.json:25]: Thats my storage website, I needed to put server files somewhere beacuse there is no official server files.

File: game_eggs/cod2/egg-call-of-duty2.json (hunk 1 above)
Comment (BacaCro) [reply game_eggs/cod2/egg-call-of-duty2.json:16]: Sorry my bad.

Comment (BacaCro) [reply game_eggs/cod2/egg-call-of-duty2.json:16]: Sorry, my bad

**Comment by BacaCro** [comment]
I'm closing this pull request, sorry for bothering :)

**Comment by QuintenQVD0** [comment]
I am going to close this because:
- an non offical image
- their are no offical dedicated servers
- the files are download from a none owner source

By their tos you can not host dedicated servers. 
So sorry but I am closing this. Thanks for your time.

**Pull request closed without merging**